packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -25,24 +30,100 @@ const notFinished: AWSCDKAsyncCustomResource.IsCompleteResponse = {
+export const isComplete = cfnResponse.safeHandler(isCompleteHandler);
 /**
- * Util function to log especially for nested objects and arrays
- * @param msg
- * @param other arrays of arguments to be logged
+ * Handler for requests sent from CloudFormation for custom resource.
+ * This is the entry point of the custom resource.
+ *
+ * Note: This is adapted from the AWS CDK's provider framework.
+ * https://github.com/aws/aws-cdk/blob/11621e7/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts
+ * @param cfnRequest Request received from CloudFormation
  */
-const log = (msg: string, ...other: any[]) => {
-  console.log(
-    msg,
-    other.map((o) => (typeof o === 'object' ? JSON.stringify(o, undefined, 2) : o)),
-  );
-};
+// eslint-disable-next-line func-style
+async function onEventHandler(cfnRequest: AWSLambda.CloudFormationCustomResourceEvent): Promise<void> {
+  const sanitizedRequest = { ...cfnRequest, ResponseURL: '...' } as const;
+  log('onEventHandler', sanitizedRequest);
+
+  cfnRequest.ResourceProperties = cfnRequest.ResourceProperties || {};
+
+  const onEventResult = await processOnEvent(sanitizedRequest);
+  log('onEvent returned:', onEventResult);
+
+  // merge the request and the result from onEvent to form the complete resource event
+  // this also performs validation.
+  const resourceEvent = createResponseEvent(cfnRequest, onEventResult);
+  log('event:', onEventResult);
+
+  if (cfnRequest.RequestType == 'Delete') {
+    // If the RequestType is `Delete`, we can submit the response to CFN.
+    // There's no need to invoke the waiter state machine.
+    log('Submitting response to CloudFormation');
+    await cfnResponse.submitResponse('SUCCESS', resourceEvent, { noEcho: resourceEvent.NoEcho });
+  } else {
+    // otherwise we start the waiter state machine to invoke
+    // `isComplete` in predefined intervals.
+    const waiter = {
+      stateMachineArn: getEnv('WAITER_STATE_MACHINE_ARN'),
+      name: resourceEvent.RequestId,
+      input: JSON.stringify(resourceEvent),
+    };
+
+    log('starting waiter', {
+      stateMachineArn: waiter.stateMachineArn,
+      name: resourceEvent.RequestId,
+    });
+
+    await startExecution(waiter);
+  }
+}
+
+/**
+ * Handler for state machine events polling completion status of resource modification.
+ *
+ * Note: This is adapted from the AWS CDK's provider framework.
+ * https://github.com/aws/aws-cdk/blob/11621e7/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts
+ * @param event isComplete request received from Waiter State Machine.
+ */
+// eslint-disable-next-line func-style
+async function isCompleteHandler(event: AWSCDKAsyncCustomResource.IsCompleteRequest): Promise<void> {
+  const sanitizedRequest = { ...event, ResponseURL: '...' } as const;
+  log('isComplete', sanitizedRequest);
+
+  const isCompleteResult = await processIsComplete(sanitizedRequest);
+  log('isComplete result', isCompleteResult);
```

Comment:
as above: do we know this result is safe to log?

Reply:
This is safe to log.  `isCompleteResult` is nothing more than:
```ts
{ isComplete: true } 
// or
{ isComplete: false }
```
[[reference](https://github.com/aws-amplify/amplify-category-api/blob/8777cd1d9609ef4d85c5ea3c95b249cc13ade6e4/packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts#L20-L25)]

It's also [currently logged by the CDK's custom resource provider framework](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts#L68).